pex/pex.py

```diff
@@ -654,20 +655,25 @@ def execute_content(
         exec_function(ast, globals_map)
         return None
 
-    @classmethod
-    def execute_entry(cls, entry_point):
+    def execute_entry(self, entry_point):
         # type: (str) -> Any
-        runner = cls.execute_pkg_resources if ":" in entry_point else cls.execute_module
-        return runner(entry_point)
+        if ":" in entry_point:
+            return self.execute_pkg_resources(entry_point)
 
-    @classmethod
-    def execute_module(cls, module_name):
-        # type: (str) -> None
-        cls.demote_bootstrap()
+        alter_sys = not zipfile.is_zipfile(self._pex)
+        return self.execute_module(entry_point, alter_sys)
+
+    def execute_module(
+        self,
+        module_name,  # type: str
+        alter_sys,  # type: bool
+    ):
+        # type: (...) -> None
+        self.demote_bootstrap()
```

Comment:
A few inline comments on why the various execution modes need to `demote` vs `alter_sys` vs etc wouldn't hurt.

Reply:
All execution modes demote so there is no vs.

I'll add verbage explaining why the zipapp case does not alter_sys though.

Reply:
Fixed.